### PR TITLE
update_ios_feature_usage_events_w_ping_parse_time_logic

### PR DESCRIPTION
**Pull request by rhythmofrain63**
Changes
1. ping parse time joining logic
2. add segments: channel, country, is_default, adjust_network

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-2643)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql:8]: Could we use `QUALIFY` here instead of the subquery?

File: sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql:75]: Remove the comment if no longer an open question.

File: sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql:77]: Remove if not needed.

File: sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/schema.yaml
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/schema.yaml:2]: We will also need `submission_date` as for the metrics feature usage table:
https://github.com/mozilla/bigquery-etl/pull/4956/files

File: sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql:280]: Remove if not needed.

File: sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql:279]: Please also add `@submission_date AS submission_date` before the `ping_date` field, we will need this for partitioning.

Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql:279]: Please also make sure to include it in the group by below.

File: sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql:36]: My guess is that we could end up with multiple records for a single user here? We should probably make sure that does not happen to avoid accidental duplication downstream.

**Review by kik-kik — APPROVED** [review]
Looks fine to me, before merging we should make sure that the default CTE does not introduce duplicates.

File: sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql
Comment (rhythmofrain63) [reply sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql:36]: @kik-kik only 391 users out of >2 million have multiple default browser records. I added a line to include only one record for each user.

Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/firefox_ios_derived/feature_usage_events_v1/query.sql:36]: Thanks!

**Review by kik-kik — APPROVED** [review]
LGTM, let's wait until tomorrow to merge as action will be required on my side otherwise we will see errors in jobs over night.

**Pull request merged**
